### PR TITLE
add metrics for recreate/update resource event when sync work status

### PR DESCRIPTION
**Pull request by chaosi-zju**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

add metrics for recreate/update resource event when sync work status

detail backgroud:

When I create a sa and a sa type secret in control plane, yet, I only propagate the secret to member clusters, then the secret will be deleted immediately by sa-controller of the member cluster. In turn the work_status_controller will watch the deletion event and recreate the resource to cluster, this cycle goes on and never stops.

Normally, there shouldn't be a large number of recreate events, so, I want to record a metric of the recreate event which can monitor the abnormal situation.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Test report:

```shell
# HELP recreate_resource_to_cluster Number of recreating operation of the resource to a target member cluster. By the result, 'error' means a resource recreated failed. Otherwise 'success'. Cluster means the target member cluster.
# TYPE recreate_resource_to_cluster counter
recreate_resource_to_cluster{apigroup="v1",cluster="member1",kind="Secret",result="success"} 810
recreate_resource_to_cluster{apigroup="v1",cluster="member2",kind="Secret",result="success"} 809
# HELP update_resource_to_cluster Number of updating operation of the resource to a target member cluster. By the result, 'error' means a resource updated failed. Otherwise 'success'. Cluster means the target member cluster.
# TYPE update_resource_to_cluster counter
update_resource_to_cluster{apigroup="rbac.authorization.k8s.io/v1",cluster="member1",kind="ClusterRole",result="success"} 2
update_resource_to_cluster{apigroup="rbac.authorization.k8s.io/v1",cluster="member2",kind="ClusterRole",result="success"} 1
```

**Does this PR introduce a user-facing change?**:

```release-note
karmada-controller-manager: add metrics for recreate/update resource event when sync work status
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @chaunceyjiang @whitewindmills would like to have a look?
/cc @chaunceyjiang  @whitewindmills

**Comment by XiShanYongYe-Chang** [comment]
How about updating the release note like this:
```
karmada-controller-manager/karmada-agent: add metrics for recreate/update resource event when sync work status
```

File: pkg/metrics/resource.go [hunk 1]
```diff
@@ -30,6 +30,8 @@ const (
 	policyApplyAttemptsMetricsName             = "policy_apply_attempts_total"
 	syncWorkDurationMetricsName                = "binding_sync_work_duration_seconds"
 	syncWorkloadDurationMetricsName            = "work_sync_workload_duration_seconds"
+	reCreateResourceToCluster                  = "re_create_resource_to_cluster"
```
Comment (XiShanYongYe-Chang) [inline pkg/metrics/resource.go:33]: recreate should be a word.

How about this: `recreate_resource_in_cluster`

File: pkg/metrics/resource.go [hunk 2]
```diff
@@ -67,6 +69,16 @@ var (
 		Buckets: prometheus.ExponentialBuckets(0.001, 2, 12),
 	}, []string{"result"})
 
+	reCreateResourceWhenSyncWorkStatus = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: reCreateResourceToCluster,
+		Help: "Number of re-creating operation of the resource to a target member cluster. By the result, 'error' means a resource recreated failed. Otherwise 'success'. Kind means the resource kind. Cluster means the target member cluster.",
```
Comment (XiShanYongYe-Chang) [inline pkg/metrics/resource.go:74]: re-creating -> recreating

File: pkg/metrics/resource.go [hunk 3]
```diff
@@ -67,6 +69,16 @@ var (
...
+	}, []string{"result", "kind", "cluster"})
```
Comment (XiShanYongYe-Chang) [inline pkg/metrics/resource.go:75]: Would it be better to replace kind with gvk?

Comment (chaosi-zju) [reply pkg/metrics/resource.go:75]: done

**Comment by XiShanYongYe-Chang** [comment]
Thanks~
/lgtm

File: pkg/metrics/resource.go [hunk 4]
```diff
@@ -67,6 +69,16 @@ var (
...
 	}, []string{"result"})
 
+	recreateResourceWhenSyncWorkStatus = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: recreateResourceToCluster,
+		Help: "Number of recreating operation of the resource to a target member cluster. By the result, 'error' means a resource recreated failed. Otherwise 'success'. GVK means the group/kind/version of the resource. Cluster means the target member cluster.",
+	}, []string{"result", "gvk", "cluster"})
+
+	updateResourceWhenSyncWorkStatus = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: updateResourceToCluster,
+		Help: "Number of updating operation of the resource to a target member cluster. By the result, 'error' means a resource recreated failed. Otherwise 'success'. GVK means the group/kind/version of the resource. Cluster means the target member cluster.",
```
Comment (whitewindmills) [inline pkg/metrics/resource.go:79]: `means a resource recreated failed` => `means a resource update failed`

File: pkg/metrics/resource.go [hunk 5]
```diff
@@ -67,6 +69,16 @@ var (
 		Buckets: prometheus.ExponentialBuckets(0.001, 2, 12),
 	}, []string{"result"})
 
+	recreateResourceWhenSyncWorkStatus = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: recreateResourceToCluster,
+		Help: "Number of recreating operation of the resource to a target member cluster. By the result, 'error' means a resource recreated failed. Otherwise 'success'. GVK means the group/kind/version of the resource. Cluster means the target member cluster.",
+	}, []string{"result", "gvk", "cluster"})
```
Comment (whitewindmills) [inline pkg/metrics/resource.go:75]: the value of `gvk` looks like `apps/v1, Kind=Deployment`. such values ​​are too complex to be used for label filtering.
`apiVersion` and `kind` are recommended.

Comment (chaosi-zju) [reply pkg/metrics/resource.go:75]: thanks

**Comment by chaosi-zju** [comment]
hi @whitewindmills, do you have any further comments?

**Comment by XiShanYongYe-Chang** [comment]
Can you help add those new metrics in the website?

**Pull request merged**